src/lib/support/IntrusiveList.h

```diff
@@ -195,6 +213,35 @@ class IntrusiveListBase
         mNode.Remove();
     }
 
+    IntrusiveListBase(const IntrusiveListBase &) = delete;
+    IntrusiveListBase & operator=(const IntrusiveListBase &) = delete;
+
+    IntrusiveListBase(IntrusiveListBase && that) : mNode(&mNode, &mNode)
+    {
+        if (!that.Empty())
+        {
+            mNode.TakePlaceOf(&that.mNode);
+            that.mNode.mNext = &that.mNode;
+            that.mNode.mPrev = &that.mNode;
+        }
```

Comment:
```suggestion
        *this = std::move(that);
```
so we don't have two copies of this code?